### PR TITLE
PKCS12 builder support

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
*Description of changes:*

Adds PKCS12 support to the MQTT311 and MQTT5 builders.

___________

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/iot/MqttClient.h [hunk 1]
```diff
@@ -100,6 +100,16 @@ namespace Aws
             friend class MqttClientConnectionConfigBuilder;
         };
 
+        /**
+         * A simple struct to hold the options for creating a PKCS12 builder. Used to differentiate the
+         * PKCS12 builder from the mTLS builder, which would use the exact sample input types without this struct.
+         */
+        struct pkcs12Options
+        {
+            const char *pkcs12_file;
```
Comment (bretambrose) [inline include/aws/iot/MqttClient.h:109]: Let's make these std::strings

File: include/aws/iot/Mqtt5Client.h [hunk 2]
```diff
@@ -173,6 +173,22 @@ namespace Aws
...
+             * @param hostName - AWS IoT endpoint to connect to
+             * @param pkcs12_file - The PKCS12 file to use
+             * @param pkcs12_password - The password for the PKCS12 file
+             * @param allocator - memory allocator to use
+             *
+             * @return Mqtt5ClientBuilder
+             */
+            static Mqtt5ClientBuilder *NewMqtt5ClientBuilderWithMtlsPkcs12(
+                const Crt::String hostName,
+                const char *pkcs12_file,
```
Comment (bretambrose) [inline include/aws/iot/Mqtt5Client.h:188]: Split file and password into a separate struct maybe?  Also, no raw c-strings

File: include/aws/iot/MqttClient.h [hunk 3]
```diff
@@ -100,6 +100,16 @@ namespace Aws
             friend class MqttClientConnectionConfigBuilder;
         };
 
+        /**
+         * A simple struct to hold the options for creating a PKCS12 builder. Used to differentiate the
+         * PKCS12 builder from the mTLS builder, which would use the exact sample input types without this struct.
+         */
+        struct pkcs12Options
```
Comment (bretambrose) [inline include/aws/iot/MqttClient.h:107]: Capitalize first letter here

File: include/aws/iot/Mqtt5Client.h (hunk 2 above)
Comment (TwistedTwigleg) [reply include/aws/iot/Mqtt5Client.h]: I moved the struct to `MqttCommon.h` and used that in both MQTT311 and MQTT5.

File: tests/MqttClientTest.cpp [hunk 4]
```diff
@@ -349,3 +357,108 @@ static int s_TestIoTMqtt311ConnectWithSigningCustomAuth(Aws::Crt::Allocator *all
...
+        aws_string_destroy(pkcs12_key);
+        aws_string_destroy(pkcs12_password);
+
+        // If in Codebuild, return as a 'success' even though it is a skip
+        if (codebuild_buildID && aws_string_is_valid(codebuild_buildID))
+        {
+            aws_string_destroy(codebuild_buildID);
+            return AWS_OP_SUCCESS;
+        }
+        aws_string_destroy(codebuild_buildID);
```
Comment (TwistedTwigleg) [inline tests/MqttClientTest.cpp:395]: Note: This is an unfortunate side effect of Codebuild counting a skipped test as a failure. This is probably something we'll need to look into further later, and I'll probably do so when we get to the C++ CRT in the unit-test/sample split work, but for now, I'm just checking for a Codebuild-specific environment variable and "passing" the test if it exists.

**Comment by TwistedTwigleg** [comment]
Thanks! Merging into `main`...

**Pull request merged**